src/units/abilities.cpp

```diff
@@ -1241,6 +1241,9 @@ std::string attack_type::select_damage_type(const unit_ability_list& damage_type
  */
 std::pair<std::string, std::string> attack_type::damage_type() const
 {
+	if(type().empty() && range().empty()){
```

Comment:
Do we have a standard way of identifying the GUI's dummy attack? Without a comment on this code, it's not clear why `type().empty() && range().empty()` would ignore damage_type abilities.

Suggested fix (added to `attack_type`)
```
/**
 * Returns true if this is a dummy attack_type, for example the placeholder that the unit_attack dialog
 * uses when a defender has to weapon for a given range.
 */
bool empty() const {
    return id.empty();
}
```

Reply:
i believe what [attack] don't have id, i prefer check if have name range and type, if one is empty, it would be a error in wml code but if the three are empty then it's a gui's dummy attack

Reply:
id was hardcoded, but never used in mainline unit, it's not a good check without other attributes

Reply:
i would use your idea to add standard check but with my own critera